src/app/components/AnalyticsConsent/initializeMatomo.ts

```diff
@@ -0,0 +1,88 @@
+/// <reference types="matomo-types" />
+
+import { redactURL } from './redact'
+
+// Generate snippet https://matomo.oasis.io/index.php?module=CoreAdminHome&action=trackingCodeGenerator&idSite=1&period=day&date=yesterday&updated=false
+// Snippet from https://developer.matomo.org/guides/tracking-javascript-guide#finding-the-piwik-tracking-code
+window._paq = window._paq || []
+export const matomoDomain = 'https://matomo.oasis.io/'
+window._paq.push(['setSiteId', '3'])
+// https://developer.matomo.org/guides/tracking-consent#step-1-require-consent
+// https://matomo.org/faq/general/faq_156/
+window._paq.push(['requireConsent'])
+window._paq.push(['setDoNotTrack', true])
+window._paq.push(['enableLinkTracking'])
+window._paq.push(['setTrackerUrl', `${matomoDomain}matomo.php`])
+
+// https://developer.matomo.org/guides/tracking-javascript-guide#tracking-one-domain-and-its-subdomains-in-the-same-website
+// https://matomo.org/faq/how-to/faq_23654/
+// window._paq.push(['setDomains', ['*.domain1.com', '*.domain2.com']])
+// window._paq.push(['enableCrossDomainLinking'])
+// window._paq.push(['setCookieDomain', '*.example.org'])
+
+window._paq.push([
+  'setCustomRequestProcessing',
+  (queryString: string) => {
+    const params = new URLSearchParams(queryString)
+    for (const name of ['action_name', 'url', 'urlref']) {
+      const v = params.get(name)
+      if (v) {
+        params.set(name, redactURL(v))
+      }
+    }
+
+    return params.toString()
+  },
+])
+
+export function loadMatomo() {
```

Comment:
Does Matomo have any "flush" feature. In a sense that, if we listen to `beforeunload` event. We could force the analytics propagation to BE, otherwise the data will be lost upon page reload.

Reply:
Doing some research around `beforeunload`, I don't see this implementation as a problem. You can make a non-blocking HTTP request and close the page. In case that HTTP request fails, the code at least tried to submit the analytics collected - and if lost, it is not such a big deal.

Reply:
Yeah, it seems to be sending tracking data after each action or after a short wait. And docs say it uses navigator.sendBeacon

https://developer.matomo.org/api-reference/tracking-javascript#advanced-uses
> disableAlwaysUseSendBeacon: navigator.sendBeacon which is enabled by default
> setRequestQueueInterval: defaults to 2.5 seconds

Reply:
In that case, seems fine.